### PR TITLE
Add a KF with a dynamic model

### PR DESCRIPTION
**Pull request by benjamin0**
Subclasses the KalmanFilter to add a StaticKalmanFilter and a DynamicKalmanFilter with a 9 state process model from Leick.

@swift-nav/estimation

### CODE REVIEW

File: tests/test_kalman_filter.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
-
 import pytest
 import numpy as np
 
-from gnss_analysis import ephemeris
 from gnss_analysis.filters import kalman_filter
 
-
+@pytest.mark.parametrize('static', [True, False])
```
Comment (akleeman) [inline tests/test_kalman_filter.py]: nit: I (think) we could clean this up a bit by doing:
```
@pytest.mark.parametrize('filter_class', [kalman_filter.KalmanFilter,
                                          kalman_filter.StaticKalmanFilter])
 def test_kalman_drops_sat(synthetic_stationary_states,
                          drop_ref, filter_class):
```
Then,
```
  expected_filter = filter_class()
  everyother_filter = filter_class()
  simultaneous_filter = filter_class()
```
and actually we could probably also parametrize the 'everyother', 'simultaneous' bit too.

File: gnss_analysis/filters/kalman_filter.py [hunk 2]
```diff
@@ -389,6 +336,206 @@ def update_matched_obs(self, rover_obs, base_obs):
...
+    such that:
+
+        x_{k|k-1} = F x_{k-1|k-1} + N(0, Q)
+    """
+    # It's possible that the observation_model will drop satellites
+    # when it detects slips, so this definition of n needs to be after
+    n = self.x.size
+    # the process model.
+    F = np.eye(n)
+    F_dynamics = np.array(np.bmat([[np.eye(self.n_dim), self.dt * np.eye(self.n_dim), 0.5 * self.dt**2 * np.eye(self.n_dim)],
```
Comment (akleeman) [inline gnss_analysis/filters/kalman_filter.py]: nit: could you describe what F looks like in a comment.

**Comment by akleeman** [comment]
Looks good!

File: tests/test_kalman_filter.py (hunk 1 above)
Comment (benjamin0) [inline tests/test_kalman_filter.py]: good call, fixed.

File: gnss_analysis/filters/kalman_filter.py (hunk 2 above)
Comment (benjamin0) [inline gnss_analysis/filters/kalman_filter.py]: done

**Pull request merged**